x/fungible/keeper/msg_server_unpause_zrc20.go

```diff
@@ -0,0 +1,59 @@
+package keeper
+
+import (
+	"context"
+
+	authoritytypes "github.com/zeta-chain/zetacore/x/authority/types"
+
+	cosmoserrors "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/zeta-chain/zetacore/x/fungible/types"
+)
+
+// UnpauseZRC20 unpauses the ZRC20 token
+func (k msgServer) UnpauseZRC20(
+	goCtx context.Context,
+	msg *types.MsgUnpauseZRC20,
+) (*types.MsgUnpauseZRC20Response, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check message validity
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
+
+	ok, err := k.GetAuthorityKeeper().IsAuthorized(ctx, msg)
+	if !ok || err != nil {
+		return nil, cosmoserrors.Wrap(authoritytypes.ErrUnauthorized, err.Error())
+	}
+
+	// iterate all foreign coins and set paused status
+	for _, zrc20 := range msg.Zrc20Addresses {
+		fc, found := k.GetForeignCoins(ctx, zrc20)
+		if !found {
+			return nil, cosmoserrors.Wrapf(types.ErrForeignCoinNotFound, "foreign coin not found %s", zrc20)
+		}
+		// Set status to unpaused
+		fc.Paused = false
+		k.SetForeignCoins(ctx, fc)
+	}
+
+	err = ctx.EventManager().EmitTypedEvent(
+		&types.EventZRC20PausedStatusUpdated{
+			MsgTypeUrl:     sdk.MsgTypeURL(&types.MsgUnpauseZRC20{}),
+			Action:         types.UpdatePausedStatusAction_UNPAUSE,
+			Zrc20Addresses: msg.Zrc20Addresses,
+			Signer:         msg.Creator,
+		},
+	)
```

Comment:
We no longer need to emit events for message as it is done automatically since sdk 0.47

Reply:
Not sure what you mean , The Pause and Unpause use the same event type with only the Action being different

Reply:
Ok but I don't think we need this event if we already emit one for pause or unpause

Reply:
The unpause TX has 
`Action:         types.UpdatePausedStatusAction_UNPAUSE,`
The pause TX has 
`Action:         types.UpdatePausedStatusAction_PAUSE,`

Are you saying that we should emit only one?